src/goImport.ts

```diff
@@ -104,3 +107,52 @@ export function addImport(arg: string) {
 		}
 	});
 }
+
+export function addImportToWorkspace(arg: string) {
+	const editor = vscode.window.activeTextEditor;
+	const selection = editor.selection;
+
+	let importPath = '';
+	if (selection.isSingleLine) {
+		// Attempt to load a partial import path based on currently selected text
+		let selectedText = editor.document.getText(selection);
+		if (selectedText.length > 0) {
+			if (!selectedText.startsWith('"')) {
+				selectedText = '"' + selectedText;
+			}
+			if (!selectedText.endsWith('"')) {
+				selectedText = selectedText + '"';
+			}
+			importPath = getImportPath(selectedText);
+		}
+	}
+
+	if (importPath === '') {
+		// Failing that use the current line
+		let selectedText = editor.document.lineAt(selection.active.line).text;
+		importPath = getImportPath(selectedText);
+	}
+
+	if (importPath === '') {
+		vscode.window.showErrorMessage('No import path to add');
+		return;
+	}
+
+	let filePath = fixDriveCasingInWindows(editor.document.fileName);
+	let fileDirPath = path.dirname(filePath);
+	let currentWorkspace = getCurrentGoWorkspaceFromGOPATH(getCurrentGoPath(), fileDirPath);
+	let globalPackagePath = path.join(currentWorkspace, importPath);
+	if (!fs.existsSync(globalPackagePath)) {
+		vscode.window.showErrorMessage('Import path not found on disk');
+		return;
+	}
+
+	const importPathUri = vscode.Uri.file(globalPackagePath);
+	const existingWorkspaceFolder = getRootWorkspaceFolder(importPathUri);
+	if (existingWorkspaceFolder !== undefined) {
+		vscode.window.showInformationMessage('Already available under ' + existingWorkspaceFolder.name);
+		return;
```

Comment:
What happens when we pass in a uri for an already existing folder to `vscode.workspace.updateWorkspaceFolders`? If `updateWorkspaceFolders` gracefully handles this case, then we dont need to do this check beforehand

Reply:
I can't remember if it did or didn't, I think my motivation was to let the user know why nothing happened. I can double check but I still think it's nice to let the user know that the workspace is already imported.

IE: they try to add `github.com/aws/aws-sdk-go/service/rds/rdsutils` after having `github.com/aws/aws-sdk-go` loaded.

In an ideal world I'd actually make the workspace navigate/expand the path but I've no idea how to do this in vscode

Reply:
if (existingWorkspaceFolder !== undefined) {
+		vscode.window.showInformationMessage('Already available under ' + existingWorkspaceFolder.name);
+		return;